roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -100,6 +109,68 @@ impl JobDeclaratorDownstream {
                                 Self::send(self_mutex.clone(), message).await.unwrap();
                             }
                             Ok(SendTo::None(_)) => (),
+                            Ok(SendTo::RelayNewMessage(JobDeclaration::SubmitSolution(
```

Comment:
if you move the logic from handle_submit_solution do not put it here but in a specific function. If possible add a test and prop test. Maybe in roles_logic_sv2? Or just leave it where it was and change it another time.

Reply:
I think this would require to make the handler trait async. Do we want to do this?

Reply:
before it was not. Btw you have several option one is to have a dedicated function that you call from here not from the handler

Reply:
What do you mean with "before it was no"?

Reply:
I mean that before your PR this logic was inside place  `handle_submit_solution` and you moved it here

Reply:
I moved as a method of JobDeclaratorDownastream. I could move it also to somehwere to roles-logic-sv2, I basically need the transaction list, the DeclareMiningJob and the SubmitSolutionJd message. I will do it tomorrow now I am tired. Let me know!

Reply:
Now it relies on the function `roles-logic-sv2::submit_solution_to_block()`.

Reply:
changed, not there is a wrapper struct in roles-log::utils and From is implemented on that struct